### DISCUSSION TITLE
How to use ContextMenu?

### DISCUSSION THREAD

**Discussion by alvahtin** [AvaloniaUI/Avalonia#15165] [Q&A] [answered]
I put ContextMenu in Canvas:

```
<Canvas>
     ........................
     <Canvas.ContextMenu>
        <ContextMenu>
          <MenuItem Header="test" />
        </ContextMenu>
      </Canvas.ContextMenu>
</Canvas>
```
But it open when clicked on the controls in Canvas (TextBlock, Lines, etc). But if I click in the free area in Canvas, it don't open.

I try put it in the Window, but it still open when clicked on the controls, but not on free area.

I want the context menu to open everywhere in the Canvas.

**Comment by jp2masa** [comment, marked as answer]
You have to set `Background="Transparent"` or any other color on the `Canvas`. It is `null` by default, which means that it's not hit test visible.

**Reply by alvahtin** [reply]
Thank you!
